clouddriver-openstack/src/main/groovy/com/netflix/spinnaker/clouddriver/openstack/deploy/ops/loadbalancer/UpsertOpenstackLoadBalancerAtomicOperation.groovy

```diff
@@ -219,13 +219,27 @@ class UpsertOpenstackLoadBalancerAtomicOperation extends AbstractOpenstackLoadBa
       task.updateStatus UPSERT_LOADBALANCER_PHASE, "Created listener $name in ${region}"
       task.updateStatus UPSERT_LOADBALANCER_PHASE, "Creating pool $name in ${region}"
       LbPoolV2 lbPool = blockingStatusChecker.execute {
-        provider.createPool(region, name, currentListener.internalProtocol.name(), algorithm.name(), listener.id)
+        provider.createPool(region, name, setInternalProtocol(currentListener.externalProtocol.name()), algorithm.name(), listener.id)
       }
       task.updateStatus UPSERT_LOADBALANCER_PHASE, "Created pool $name in ${region}"
       updateHealthMonitor(region, loadBalancerId, lbPool, healthMonitor)
     }
   }
 
+  private static String setInternalProtocol(String externalProtocol) {
```

Comment:
This would work for 

```
protected void addListenersAndPools(String region, String loadBalancerId, String name, Algorithm algorithm, Map<String, Listener> listeners, HealthMonitor healthMonitor) {
    BlockingStatusChecker blockingStatusChecker = createBlockingActiveStatusChecker(region, loadBalancerId)
    listeners?.each { String key, Listener currentListener ->
      task.updateStatus UPSERT_LOADBALANCER_PHASE, "Creating listener $name in ${region}"
      ListenerV2 listener = blockingStatusChecker.execute {
        provider.createListener(region, name, currentListener.externalProtocol.name(), currentListener.externalPort, key, loadBalancerId)
      }
      task.updateStatus UPSERT_LOADBALANCER_PHASE, "Created listener $name in ${region}"
      task.updateStatus UPSERT_LOADBALANCER_PHASE, "Creating pool $name in ${region}"
      LbPoolV2 lbPool = blockingStatusChecker.execute {
        println 'currentListener: ' + currentListener
        provider.createPool(region, name, currentListener.externalProtocol.internalProtocol, algorithm.name(), listener.id)
      }
      task.updateStatus UPSERT_LOADBALANCER_PHASE, "Created pool $name in ${region}"
      updateHealthMonitor(region, loadBalancerId, lbPool, healthMonitor)
    }
  }```

But updateListenersAndPools uses ListenerV2. 
```
protected void updateListenersAndPools(String region, String loadBalancerId, Algorithm algorithm, Collection<ListenerV2> listeners, HealthMonitor healthMonitor) {
    BlockingStatusChecker blockingStatusChecker = createBlockingActiveStatusChecker(region, loadBalancerId)
    listeners?.each { ListenerV2 currentListener ->
      LbPoolV2 lbPool = null

      if (currentListener.defaultPoolId) {
        lbPool = provider.getPool(region, currentListener.defaultPoolId)
        if (lbPool.lbMethod.name() != algorithm.name()) {
          task.updateStatus UPSERT_LOADBALANCER_PHASE, "Updating pool $lbPool.name in ${region} ..."
          blockingStatusChecker.execute { provider.updatePool(region, lbPool.id, algorithm.name()) }
          task.updateStatus UPSERT_LOADBALANCER_PHASE, "Updated pool $lbPool.name in ${region}."
        }
      } else {
        Map<String, String> listenerMap = parseListenerKey(currentListener.description)
        if (!listenerMap?.isEmpty()) {
          println 'listenerMap: ' + listenerMap // [externalProtocol:HTTP, externalPort:80, internalPort:8080]
          // Create a pool
          task.updateStatus UPSERT_LOADBALANCER_PHASE, "Creating pool $currentListener.name in ${region} ..."
          lbPool = blockingStatusChecker.execute {
            provider.createPool(region, currentListener.name, listenerMap.externalProtocol.internalProtocol, algorithm.name(), currentListener.id)
          }
          task.updateStatus UPSERT_LOADBALANCER_PHASE, "Created pool $currentListener.name in ${region}."
        }
      }

      updateHealthMonitor(region, loadBalancerId, lbPool, healthMonitor)
    }
  }
```

should we not be using ListenerV2? be mapping it to Listener somehow? or continue using the setInternalProtocol for updateListenersAndPools? Do we need both?